API/Helpers/query.js

```diff
@@ -24,3 +27,14 @@ export const createMemory = (title, story, mood, picture) => {
     });
 }
 
+export const updateMemory = (id, title, story, mood, picture) => {
+    const result = memories.findIndex(memory => memory.id === +id);
+    memories[result] = {
+        id: id,
+        title: title,
+        mood: mood,
+        picture: picture,
+        story: story,
+        date: new Date()
```

Comment:
how about if I want to make partial updates, the user doesn't have to supply every information. Consider using the [null coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)
```suggestion
        id: id,
        title: title ?? memories[result].title,
        mood: mood ?? memories[result].mood ,
        picture: picture ?? memories[result].picture,
        story: story ?? memories[result].story,
        date: new Date() ?? memories[result].date
```